Organize Imports: Debug Failure. Did not expect InterfaceDeclaration to have an Identifier in its trivia

From https://github.com/Microsoft/vscode/issues/46698

**TypeScript Version:**  2.9.0-dev.20180327

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** 
- Organize imports

**Code**

1. In the vscode codebase
2. Open `modes.ts`: https://github.com/Microsoft/vscode/blob/master/src/vs/editor/common/modes.ts
3. Delete the `{ IMarkerData }` import
4. Run Organize imports

**Bug:**
Debug failure:

```
[Error  - 13:54:12] 'organizeImports' request failed with error.
Error processing request. Debug Failure. Did not expect InterfaceDeclaration to have an Identifier in its trivia
Error: Debug Failure. Did not expect InterfaceDeclaration to have an Identifier in its trivia
    at NodeObject.addSyntheticNodes (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:83936:34)
    at processNode (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:83980:29)
    at NodeObject.createChildren (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:83994:21)
    at NodeObject.getChildren (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:84020:22)
    at getTokenAtPositionWorker (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:66397:43)
    at getTouchingToken (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:66384:16)
    at Object.getTouchingPropertyName (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:66380:16)
    at /Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:71736:47
    at Array.some (native)
    at Object.isSymbolReferencedInFile (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:71735:85)
    at isDeclarationUsed (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:74069:102)
    at /Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:74055:95
    at Array.filter (native)
    at removeUnusedImports (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:74055:66)
    at /Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:74006:43
    at Object.flatMap (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:1774:25)
    at organizeImportsWorker (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:74004:41)
    at Object.organizeImports (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:73989:13)
    at Object.organizeImports (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:85310:39)
    at IOSession.Session.organizeImports (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:92327:60)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:91225:61)
    at /Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:92510:88
    at IOSession.Session.executeWithRequestId (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:92501:28)
    at IOSession.Session.executeCommand (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:92510:33)
    at IOSession.Session.onMessage (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:92530:35)
    at Interface.<anonymous> (/Users/matb/projects/vscode/node_modules/typescript/lib/tsserver.js:93738:27)
    at emitOne (events.js:96:13)
    at Interface.emit (events.js:191:7)
    at Interface._onLine (readline.js:241:10)
    at Interface._normalWrite (readline.js:384:12)
    at Socket.ondata (readline.js:101:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
```

Also occurs on `references`

Log:
```
Info 0    [11:8:54.520] Starting TS Server
Info 1    [11:8:54.520] Version: 2.9.0-dev.20180325
Info 2    [11:8:54.520] Arguments: /usr/share/code-insiders/code-insiders /home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName /tmp/vscode-tscancellation-bbceac93f5f306f17722.sock* --logVerbosity verbose --logFile /home/andy/.config/Code - Insiders/logs/20180328T105646/vscode.typescript-language-features_6/tsserver-log-6evHOS/tsserver.log --locale en
Info 3    [11:8:54.520] Platform: linux NodeVersion: 7 CaseSensitive: true
Info 4    [11:8:54.522] Provided types map file "/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js/../typesMap.json" doesn't exist
Info 5    [11:8:54.523] Binding...
Info 6    [11:8:54.529] event:
    {"seq":0,"type":"event","event":"typingsInstallerPid","body":{"pid":18792}}
Info 7    [11:8:54.530] request:
    {"seq":0,"type":"request","command":"configure","arguments":{"hostInfo":"vscode"}}
Info 8    [11:8:54.531] Host information vscode
Info 9    [11:8:54.531] response:
    {"seq":0,"type":"response","command":"configure","request_seq":0,"success":true}
Perf 10   [11:8:54.531] 0::configure: async elapsed time (in milliseconds) 0.6122
Info 11   [11:8:54.531] request:
    {"seq":1,"type":"request","command":"compilerOptionsForInferredProjects","arguments":{"options":{"module":"commonjs","target":"es2016","jsx":"preserve","allowJs":true,"allowSyntheticDefaultImports":true,"allowNonTsExtensions":true}}}
Info 12   [11:8:54.532] Scheduled: *ensureProjectForOpenFiles*
Perf 13   [11:8:54.532] 1::compilerOptionsForInferredProjects: elapsed time (in milliseconds) 0.9073
Info 14   [11:8:54.532] response:
    {"seq":0,"type":"response","command":"compilerOptionsForInferredProjects","request_seq":1,"success":true,"body":true}
Info 15   [11:8:54.533] request:
    {"seq":2,"type":"request","command":"open","arguments":{"file":"/home/andy/sample/ts/src/a.ts","fileContent":"import { IMarkerData } from 'vs/platform/markers/common/markers';\nimport b from 'b';\nimport c from 'c';\n\n/** @internal */\nexport class LanguageIdentifier {}\n","scriptKindName":"TS","projectRootPath":"/home/andy/sample/ts"}}
Info 16   [11:8:54.534] Search path: /home/andy/sample/ts/src
Info 17   [11:8:54.534] ConfigFilePresence:: Current Watches: :: File: /home/andy/sample/ts/src/tsconfig.json Currently impacted open files: RootsOfInferredProjects:  OtherOpenFiles: /home/andy/sample/ts/src/a.ts Status: File added to open files impacted by this config file
Info 18   [11:8:54.535] ConfigFilePresence:: Current Watches: :: File: /home/andy/sample/ts/src/jsconfig.json Currently impacted open files: RootsOfInferredProjects:  OtherOpenFiles: /home/andy/sample/ts/src/a.ts Status: File added to open files impacted by this config file
Info 19   [11:8:54.535] ConfigFilePresence:: Current Watches: :: File: /home/andy/sample/ts/tsconfig.json Currently impacted open files: RootsOfInferredProjects:  OtherOpenFiles: /home/andy/sample/ts/src/a.ts Status: File added to open files impacted by this config file
Info 20   [11:8:54.535] For info: /home/andy/sample/ts/src/a.ts :: Config file name: /home/andy/sample/ts/tsconfig.json
Info 21   [11:8:54.547] Opened configuration file /home/andy/sample/ts/tsconfig.json
Info 22   [11:8:54.550] FileWatcher:: Added:: WatchInfo: /home/andy/sample/ts/tsconfig.json 2000 Project: /home/andy/sample/ts/tsconfig.json WatchType: Config file for the program
Info 23   [11:8:54.550] DirectoryWatcher:: Added:: WatchInfo: /home/andy/sample/ts 1 Project: /home/andy/sample/ts/tsconfig.json WatchType: Wild card directory
Info 24   [11:8:54.558] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 25   [11:8:54.580] DirectoryWatcher:: Added:: WatchInfo: /home/andy/sample/ts 1 Project: /home/andy/sample/ts/tsconfig.json WatchType: Directory of Failed lookup locations in module resolution
Info 26   [11:8:54.593] FileWatcher:: Added:: WatchInfo: /home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/lib.d.ts 500 Project:  WatchType: Closed Script info
Info 27   [11:8:55.43] DirectoryWatcher:: Added:: WatchInfo: /home/andy/sample/ts/node_modules/@types 1 Project: /home/andy/sample/ts/tsconfig.json WatchType: Type root directory
Info 28   [11:8:55.44] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 1 structureChanged: true Elapsed: 486ms
Info 29   [11:8:55.46] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 30   [11:8:55.49] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 3ms
Info 31   [11:8:55.50] event:
    {"seq":0,"type":"event","event":"telemetry","body":{"telemetryEventName":"projectInfo","payload":{"projectId":"b9db5ecb5129c341352ecb6b5fe49524","fileStats":{"js":0,"jsx":0,"ts":1,"tsx":0,"dts":1},"compilerOptions":{},"typeAcquisition":{"enable":false,"include":false,"exclude":false},"extends":false,"files":false,"include":false,"exclude":false,"compileOnSave":false,"configFileName":"tsconfig.json","projectType":"configured","languageServiceEnabled":true,"version":"2.9.0-dev.20180325"}}}
Info 32   [11:8:55.50] event:
    {"seq":0,"type":"event","event":"configFileDiag","body":{"triggerFile":"/home/andy/sample/ts/src/a.ts","configFile":"/home/andy/sample/ts/tsconfig.json","diagnostics":[]}}
Info 33   [11:8:55.51] Project '/home/andy/sample/ts/tsconfig.json' (Configured) 0
Info 33   [11:8:55.51] 	Files (2)
	/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/lib.d.ts
	/home/andy/sample/ts/src/a.ts

Info 33   [11:8:55.51] -----------------------------------------------
Info 33   [11:8:55.51] Open files: 
Info 33   [11:8:55.51] 	FileName: /home/andy/sample/ts/src/a.ts ProjectRootPath: /home/andy/sample/ts
Info 33   [11:8:55.51] 		Projects: /home/andy/sample/ts/tsconfig.json
Perf 33   [11:8:55.51] 2::open: async elapsed time (in milliseconds) 518.4259
Info 34   [11:8:55.51] request:
    {"seq":3,"type":"request","command":"getSupportedCodeFixes","arguments":null}
Perf 35   [11:8:55.51] 3::getSupportedCodeFixes: elapsed time (in milliseconds) 0.2293
Info 36   [11:8:55.52] response:
    {"seq":0,"type":"response","command":"getSupportedCodeFixes","request_seq":3,"success":true,"body":["1002","1003","1005","1006","1009","1010","1012","1014","1015","1016","1017","1018","1019","1020","1021","1022","1023","1024","1028","1029","1030","1031","1034","1035","1036","1038","1039","1040","1041","1042","1043","1044","1045","1046","1047","1048","1049","1051","1052","1053","1054","1055","1056","1057","1058","1059","1060","1061","1062","1063","1064","1066","1068","1070","1071","1079","1084","1085","1086","1089","1090","1091","1092","1093","1094","1095","1096","1097","1098","1099","1100","1101","1102","1103","1104","1105","1107","1108","1109","1110","1113","1114","1115","1116","1117","1118","1119","1120","1121","1122","1123","1124","1125","1126","1127","1128","1129","1130","1131","1132","1134","1135","1136","1137","1138","1139","1140","1141","1142","1144","1146","1147","1148","1149","1150","1155","1156","1157","1160","1161","1162","1163","1164","1165","1166","1168","1169","1170","1171","1172","1173","1174","1175","1176","1177","1178","1179","1180","1181","1182","1183","1184","1185","1186","1187","1188","1189","1190","1191","1192","1193","1194","1196","1197","1198","1199","1200","1202","1203","1205","1206","1207","1208","1209","1210","1211","1212","1213","1214","1215","1216","1218","1219","1220","1221","1222","1223","1224","1225","1226","1227","1228","1229","1230","1231","1232","1233","1234","1235","1236","1237","1238","1239","1240","1241","1242","1243","1244","1245","1246","1247","1248","1249","1250","1251","1252","1253","1254","1255","1300","1308","1312","1313","1314","1315","1316","1317","1318","1319","1320","1321","1322","1323","1324","1325","1326","1327","1328","1329","1330","1331","1332","1333","1334","1335","1336","1337","1338","2300","2301","2302","2303","2304","2305","2306","2307","2308","2309","2310","2311","2312","2313","2314","2315","2316","2317","2318","2319","2320","2321","2322","2323","2324","2325","2326","2327","2328","2329","2330","2331","2332","2333","2334","2335","2336","2337","2338","2339","2340","2341","2342","2343","2344","2345","2346","2347","2348","2349","2350","2351","2352","2353","2354","2355","2356","2357","2358","2359","2360","2361","2362","2363","2364","2365","2366","2368","2369","2370","2371","2372","2373","2374","2375","2376","2377","2378","2379","2380","2381","2382","2383","2384","2385","2386","2387","2388","2389","2390","2391","2392","2393","2394","2395","2396","2397","2399","2400","2401","2402","2403","2404","2405","2406","2407","2408","2409","2410","2411","2412","2413","2414","2415","2416","2417","2420","2422","2423","2424","2425","2426","2427","2428","2430","2431","2432","2433","2434","2435","2436","2437","2438","2439","2440","2441","2442","2443","2444","2445","2446","2447","2448","2449","2450","2451","2452","2453","2454","2455","2456","2457","2458","2459","2460","2461","2462","2463","2464","2465","2466","2467","2468","2469","2470","2471","2472","2473","2474","2475","2476","2477","2478","2479","2480","2481","2483","2484","2487","2488","2489","2490","2491","2492","2493","2494","2495","2496","2497","2498","2499","2500","2501","2502","2503","2504","2505","2506","2507","2508","2509","2510","2511","2512","2513","2514","2515","2516","2517","2518","2519","2520","2521","2522","2523","2524","2525","2526","2527","2528","2529","2530","2531","2532","2533","2534","2535","2536","2537","2538","2539","2540","2541","2542","2543","2544","2545","2546","2547","2548","2549","2550","2551","2552","2553","2554","2555","2556","2557","2558","2559","2560","2561","2562","2563","2564","2565","2566","2567","2600","2601","2602","2603","2604","2605","2606","2607","2608","2609","2649","2651","2652","2653","2654","2656","2657","2658","2659","2660","2661","2662","2663","2664","2665","2666","2667","2668","2669","2670","2671","2672","2673","2674","2675","2676","2677","2678","2679","2680","2681","2682","2683","2684","2685","2686","2687","2688","2689","2691","2692","2693","2694","2695","2696","2697","2698","2699","2700","2701","2702","2703","2704","2705","2706","2707","2708","2709","2710","2711","2712","2713","2714","2715","2716","2717","2718","2719","2720","2721","2722","2723","2724","4000","4002","4004","4006","4008","4010","4012","4014","4016","4019","4020","4022","4023","4024","4025","4026","4027","4028","4029","4030","4031","4032","4033","4034","4035","4036","4037","4038","4039","4040","4041","4042","4043","4044","4045","4046","4047","4048","4049","4050","4051","4052","4053","4054","4055","4056","4057","4058","4059","4060","4061","4062","4063","4064","4065","4066","4067","4068","4069","4070","4071","4072","4073","4074","4075","4076","4077","4078","4081","4082","4083","4090","4091","4092","4094","4095","4096","4097","4098","4099","4100","4101","4102","5001","5009","5010","5012","5014","5023","5024","5033","5042","5047","5051","5052","5053","5054","5055","5056","5057","5058","5059","5060","5061","5062","5063","5064","5065","5066","5067","5068","6044","6045","6046","6048","6049","6050","6051","6053","6054","6059","6064","6082","6103","6114","6131","6133","6137","6138","6140","6142","6188","6189","6192","7005","7006","7008","7009","7010","7011","7013","7015","7016","7017","7018","7019","7020","7022","7023","7024","7025","7026","7027","7028","7029","7030","7031","7032","7033","7034","7035","7036","7038","7039","8000","8001","8002","8003","8004","8005","8006","8007","8008","8009","8010","8011","8012","8013","8015","8016","8017","8018","8020","8021","8022","8023","8024","8025","8026","8027","8028","8029","9002","9003","9004","17000","17001","17002","17003","17004","17005","17006","17007","17008","17009","17010","17011","17012","17013","17014","17015","17016","17017","18000","18001","18002","18003","80001","80002","80003","80004"]}
Info 37   [11:8:55.52] Running: *ensureProjectForOpenFiles*
Info 38   [11:8:55.52] Structure before ensureProjectForOpenFiles:
Info 39   [11:8:55.52] Project '/home/andy/sample/ts/tsconfig.json' (Configured) 0
Info 39   [11:8:55.52] 	Files (2)
	/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/lib.d.ts
	/home/andy/sample/ts/src/a.ts

Info 39   [11:8:55.52] -----------------------------------------------
Info 39   [11:8:55.52] Open files: 
Info 39   [11:8:55.52] 	FileName: /home/andy/sample/ts/src/a.ts ProjectRootPath: /home/andy/sample/ts
Info 39   [11:8:55.52] 		Projects: /home/andy/sample/ts/tsconfig.json
Info 39   [11:8:55.52] Structure after ensureProjectForOpenFiles:
Info 40   [11:8:55.52] Project '/home/andy/sample/ts/tsconfig.json' (Configured) 0
Info 40   [11:8:55.52] 	Files (2)
	/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/lib.d.ts
	/home/andy/sample/ts/src/a.ts

Info 40   [11:8:55.52] -----------------------------------------------
Info 40   [11:8:55.52] Open files: 
Info 40   [11:8:55.52] 	FileName: /home/andy/sample/ts/src/a.ts ProjectRootPath: /home/andy/sample/ts
Info 40   [11:8:55.52] 		Projects: /home/andy/sample/ts/tsconfig.json
Info 40   [11:8:55.53] got projects updated in background, updating diagnostics for /home/andy/sample/ts/src/a.ts
Info 41   [11:8:55.53] event:
    {"seq":0,"type":"event","event":"projectsUpdatedInBackground","body":{"openFiles":["/home/andy/sample/ts/src/a.ts"]}}
Info 42   [11:8:55.56] request:
    {"seq":4,"type":"request","command":"getOutliningSpans","arguments":{"file":"/home/andy/sample/ts/src/a.ts"}}
Perf 43   [11:8:55.60] 4::getOutliningSpans: elapsed time (in milliseconds) 4.1175
Info 44   [11:8:55.61] response:
    {"seq":0,"type":"response","command":"getOutliningSpans","request_seq":4,"success":true,"body":[{"textSpan":{"start":{"line":5,"offset":1},"end":{"line":5,"offset":17}},"hintSpan":{"start":{"line":5,"offset":1},"end":{"line":5,"offset":17}},"bannerText":"...","autoCollapse":false},{"textSpan":{"start":{"line":6,"offset":32},"end":{"line":6,"offset":35}},"hintSpan":{"start":{"line":6,"offset":1},"end":{"line":6,"offset":35}},"bannerText":"...","autoCollapse":false}]}
Info 45   [11:8:55.61] request:
    {"seq":5,"type":"request","command":"geterr","arguments":{"delay":0,"files":["/home/andy/sample/ts/src/a.ts"]}}
Perf 46   [11:8:55.62] 5::geterr: async elapsed time (in milliseconds) 0.5108
Info 47   [11:8:55.63] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 48   [11:8:55.63] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 0ms
Info 49   [11:8:55.64] event:
    {"seq":0,"type":"event","event":"syntaxDiag","body":{"file":"/home/andy/sample/ts/src/a.ts","diagnostics":[]}}
Info 50   [11:8:55.64] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 51   [11:8:55.64] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 0ms
Info 52   [11:8:55.74] event:
    {"seq":0,"type":"event","event":"semanticDiag","body":{"file":"/home/andy/sample/ts/src/a.ts","diagnostics":[{"start":{"line":1,"offset":29},"end":{"line":1,"offset":65},"text":"Cannot find module 'vs/platform/markers/common/markers'.","code":2307,"category":"error"},{"start":{"line":2,"offset":15},"end":{"line":2,"offset":18},"text":"Cannot find module 'b'.","code":2307,"category":"error"},{"start":{"line":3,"offset":15},"end":{"line":3,"offset":18},"text":"Cannot find module 'c'.","code":2307,"category":"error"}]}}
Info 53   [11:8:55.75] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 54   [11:8:55.75] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 0ms
Info 55   [11:8:55.75] event:
    {"seq":0,"type":"event","event":"suggestionDiag","body":{"file":"/home/andy/sample/ts/src/a.ts","diagnostics":[]}}
Info 56   [11:8:55.75] event:
    {"seq":0,"type":"event","event":"requestCompleted","body":{"request_seq":5}}
Info 57   [11:8:55.293] request:
    {"seq":6,"type":"request","command":"getApplicableRefactors","arguments":{"file":"/home/andy/sample/ts/src/a.ts","startLine":1,"startOffset":1,"endLine":2,"endOffset":1}}
Info 58   [11:8:55.294] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 59   [11:8:55.294] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 0ms
Perf 60   [11:8:55.319] 6::getApplicableRefactors: elapsed time (in milliseconds) 25.8673
Info 61   [11:8:55.319] response:
    {"seq":0,"type":"response","command":"getApplicableRefactors","request_seq":6,"success":true,"body":[]}
Info 62   [11:8:55.320] request:
    {"seq":7,"type":"request","command":"configure","arguments":{"file":"/home/andy/sample/ts/src/a.ts","formatOptions":{"tabSize":4,"indentSize":4,"convertTabsToSpaces":true,"newLineCharacter":"\n","insertSpaceAfterCommaDelimiter":true,"insertSpaceAfterConstructor":false,"insertSpaceAfterSemicolonInForStatements":true,"insertSpaceBeforeAndAfterBinaryOperators":true,"insertSpaceAfterKeywordsInControlFlowStatements":true,"insertSpaceAfterFunctionKeywordForAnonymousFunctions":true,"insertSpaceBeforeFunctionParenthesis":false,"insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis":false,"insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets":false,"insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces":true,"insertSpaceAfterOpeningAndBeforeClosingTemplateStringBraces":false,"insertSpaceAfterOpeningAndBeforeClosingJsxExpressionBraces":false,"insertSpaceAfterTypeAssertion":false,"placeOpenBraceOnNewLineForFunctions":false,"placeOpenBraceOnNewLineForControlBlocks":false}}}
Info 63   [11:8:55.321] Host configuration update for file /home/andy/sample/ts/src/a.ts
Info 64   [11:8:55.321] response:
    {"seq":0,"type":"response","command":"configure","request_seq":7,"success":true}
Perf 65   [11:8:55.321] 7::configure: async elapsed time (in milliseconds) 0.5259
Info 66   [11:8:55.321] request:
    {"seq":8,"type":"request","command":"getCodeFixes","arguments":{"file":"/home/andy/sample/ts/src/a.ts","startLine":1,"startOffset":29,"endLine":1,"endOffset":65,"errorCodes":[2307]}}
Info 67   [11:8:55.322] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 68   [11:8:55.322] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 0ms
Perf 69   [11:8:55.322] 8::getCodeFixes: elapsed time (in milliseconds) 1.0995
Info 70   [11:8:55.322] response:
    {"seq":0,"type":"response","command":"getCodeFixes","request_seq":8,"success":true,"body":[]}
Info 71   [11:8:55.427] request:
    {"seq":9,"type":"request","command":"getApplicableRefactors","arguments":{"file":"/home/andy/sample/ts/src/a.ts","startLine":1,"startOffset":1,"endLine":2,"endOffset":1}}
Info 72   [11:8:55.427] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 73   [11:8:55.427] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 0ms
Perf 74   [11:8:55.428] 9::getApplicableRefactors: elapsed time (in milliseconds) 0.7572
Info 75   [11:8:55.428] response:
    {"seq":0,"type":"response","command":"getApplicableRefactors","request_seq":9,"success":true,"body":[]}
Info 76   [11:8:55.428] request:
    {"seq":10,"type":"request","command":"getCodeFixes","arguments":{"file":"/home/andy/sample/ts/src/a.ts","startLine":1,"startOffset":29,"endLine":1,"endOffset":65,"errorCodes":[2307]}}
Info 77   [11:8:55.428] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 78   [11:8:55.428] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 2 structureChanged: false Elapsed: 0ms
Perf 79   [11:8:55.428] 10::getCodeFixes: elapsed time (in milliseconds) 0.3084
Info 80   [11:8:55.428] response:
    {"seq":0,"type":"response","command":"getCodeFixes","request_seq":10,"success":true,"body":[]}
Info 81   [11:8:55.570] request:
    {"seq":11,"type":"request","command":"change","arguments":{"file":"/home/andy/sample/ts/src/a.ts","line":1,"offset":1,"endLine":2,"endOffset":1,"insertString":""}}
Perf 82   [11:8:55.571] 11::change: async elapsed time (in milliseconds) 1.3773
Info 83   [11:8:55.765] request:
    {"seq":12,"type":"request","command":"getOutliningSpans","arguments":{"file":"/home/andy/sample/ts/src/a.ts"}}
Err 84    [11:8:55.807] Exception on executing command {"seq":12,"type":"request","command":"getOutliningSpans","arguments":{"file":"/home/andy/sample/ts/src/a.ts"}}:

    Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia

    Error: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia
    at NodeObject.addSyntheticNodes (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83584:34)
    at processNode (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83628:29)
    at NodeObject.createChildren (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83642:21)
    at NodeObject.getChildren (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83668:22)
    at Object.findChildOfKind (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:66222:26)
    at spanForNode (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74169:36)
    at getOutliningSpanForNode (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74155:28)
    at walk (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74052:28)
    at visitNodes (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:13352:30)
    at Object.forEachChild (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:13559:24)
    at SourceFileObject.NodeObject.forEachChild (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83692:23)
    at addNodeOutliningSpans (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74045:24)
    at Object.collectElements (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74038:13)
    at Object.getOutliningSpans (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:84735:50)
    at IOSession.Session.getOutliningSpans (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:91366:45)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:90538:61)
    at /home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92010:88
    at IOSession.Session.executeWithRequestId (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92001:28)
    at IOSession.Session.executeCommand (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92010:33)
    at IOSession.Session.onMessage (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92030:35)
    at Interface.<anonymous> (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:93238:27)
    at emitOne (events.js:96:13)
    at Interface.emit (events.js:191:7)
    at Interface._onLine (readline.js:241:10)
    at Interface._normalWrite (readline.js:384:12)
    at Socket.ondata (readline.js:101:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
Info 85   [11:8:55.807] response:
    {"seq":0,"type":"response","command":"getOutliningSpans","request_seq":12,"success":false,"message":"Error processing request. Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia\nError: Debug Failure. Did not expect ClassDeclaration to have an Identifier in its trivia\n    at NodeObject.addSyntheticNodes (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83584:34)\n    at processNode (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83628:29)\n    at NodeObject.createChildren (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83642:21)\n    at NodeObject.getChildren (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83668:22)\n    at Object.findChildOfKind (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:66222:26)\n    at spanForNode (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74169:36)\n    at getOutliningSpanForNode (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74155:28)\n    at walk (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74052:28)\n    at visitNodes (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:13352:30)\n    at Object.forEachChild (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:13559:24)\n    at SourceFileObject.NodeObject.forEachChild (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:83692:23)\n    at addNodeOutliningSpans (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74045:24)\n    at Object.collectElements (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:74038:13)\n    at Object.getOutliningSpans (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:84735:50)\n    at IOSession.Session.getOutliningSpans (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:91366:45)\n    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:90538:61)\n    at /home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92010:88\n    at IOSession.Session.executeWithRequestId (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92001:28)\n    at IOSession.Session.executeCommand (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92010:33)\n    at IOSession.Session.onMessage (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:92030:35)\n    at Interface.<anonymous> (/home/andy/.nvm/versions/node/v9.9.0/lib/node_modules/typescript/lib/tsserver.js:93238:27)\n    at emitOne (events.js:96:13)\n    at Interface.emit (events.js:191:7)\n    at Interface._onLine (readline.js:241:10)\n    at Interface._normalWrite (readline.js:384:12)\n    at Socket.ondata (readline.js:101:10)\n    at emitOne (events.js:96:13)\n    at Socket.emit (events.js:191:7)\n    at readableAddChunk (_stream_readable.js:178:18)\n    at Socket.Readable.push (_stream_readable.js:136:10)\n    at Pipe.onread (net.js:560:20)"}
Info 86   [11:8:55.869] request:
    {"seq":13,"type":"request","command":"geterr","arguments":{"delay":0,"files":["/home/andy/sample/ts/src/a.ts"]}}
Perf 87   [11:8:55.870] 13::geterr: async elapsed time (in milliseconds) 0.5684
Info 88   [11:8:55.872] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 89   [11:8:55.882] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 3 structureChanged: true Elapsed: 10ms
Info 90   [11:8:55.883] event:
    {"seq":0,"type":"event","event":"syntaxDiag","body":{"file":"/home/andy/sample/ts/src/a.ts","diagnostics":[]}}
Info 91   [11:8:55.883] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 92   [11:8:55.884] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 3 structureChanged: false Elapsed: 0ms
Info 93   [11:8:55.887] event:
    {"seq":0,"type":"event","event":"semanticDiag","body":{"file":"/home/andy/sample/ts/src/a.ts","diagnostics":[{"start":{"line":1,"offset":15},"end":{"line":1,"offset":18},"text":"Cannot find module 'b'.","code":2307,"category":"error"},{"start":{"line":2,"offset":15},"end":{"line":2,"offset":18},"text":"Cannot find module 'c'.","code":2307,"category":"error"}]}}
Info 94   [11:8:55.888] Starting updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json
Info 95   [11:8:55.889] Finishing updateGraphWorker: Project: /home/andy/sample/ts/tsconfig.json Version: 3 structureChanged: false Elapsed: 0ms
Info 96   [11:8:55.890] event:
    {"seq":0,"type":"event","event":"suggestionDiag","body":{"file":"/home/andy/sample/ts/src/a.ts","diagnostics":[]}}
Info 97   [11:8:55.890] event:
    {"seq":0,"type":"event","event":"requestCompleted","body":{"request_seq":13}}
```

--- a/src/compiler/parser.ts
+++ b/src/compiler/parser.ts
@@ -6965,7 +6965,7 @@ namespace ts {
                 forEachChild(node, visitNode, visitArray);
                 if (hasJSDocNodes(node)) {
                     for (const jsDocComment of node.jsDoc) {
-                        forEachChild(jsDocComment, visitNode, visitArray);
+                        visitNode(<IncrementalNode><Node>jsDocComment);
                     }
                 }
                 checkNodePositions(node, aggressiveChecks);
@@ -7071,10 +7071,16 @@ namespace ts {
         function checkNodePositions(node: Node, aggressiveChecks: boolean) {
             if (aggressiveChecks) {
                 let pos = node.pos;
-                forEachChild(node, child => {
+                const visitNode = (child: Node) => {
                     Debug.assert(child.pos >= pos);
                     pos = child.end;
-                });
+                };
+                if (hasJSDocNodes(node)) {
+                    for (const jsDocComment of node.jsDoc) {
+                        visitNode(jsDocComment);
+                    }
+                }
+                forEachChild(node, visitNode);
                 Debug.assert(pos <= node.end);
             }
         }
@@ -7112,7 +7118,11 @@ namespace ts {
                     // Adjust the pos or end (or both) of the intersecting element accordingly.
                     adjustIntersectingElement(child, changeStart, changeRangeOldEnd, changeRangeNewEnd, delta);
                     forEachChild(child, visitNode, visitArray);
-
+                    if (hasJSDocNodes(child)) {
+                        for (const jsDocComment of child.jsDoc) {
+                            visitNode(<IncrementalNode><Node>jsDocComment);
+                        }
+                    }
                     checkNodePositions(child, aggressiveChecks);
                     return;
                 }

--- a/tests/cases/fourslash/incrementalParsingWithJsDoc.ts
+++ b/tests/cases/fourslash/incrementalParsingWithJsDoc.ts
@@ -1,0 +1,15 @@
+/// <reference path="fourslash.ts"/>
+
+////import a from 'a/aaaaaaa/aaaaaaa/aaaaaa/aaaaaaa';
+/////**/import b from 'b';
+////import c from 'c';
+////
+////[|/** @internal */|]
+////export class LanguageIdentifier[| { }|]
+
+// Force a syntax tree ot be created.
+verify.outliningSpansInCurrentFile(test.ranges());
+goTo.marker("");
+edit.backspace(test.marker("").position);
+verify.outliningSpansInCurrentFile(test.ranges());
+

